### ISSUE TITLE
RequiredAttribute as Key?

### ISSUE THREAD

**Issue by ShadyHarbinger** [ffhighwind/DapperExtraCRUD#1]
SqlTypeInfo evaluates RequiredAttribute as key. Therefore tha package cannot be used where a table contains required (not null) field. Reuired is not equal to Key. The real key fields which are named ID or ends with "ID" are not recognized and the created inserts etc. SQL statements includes the real key fields.
Fields probably should be checked for "ID" even if keys list is not empty. (Remove `if (keys.Count == 0)` at line 164 in SqlTypeInfo.cs?)

**Comment by ffhighwind** [comment]
Hi Shady. I was intending to mimic how [DapperSimpleCRUD ](https://github.com/ericdc1/Dapper.SimpleCRUD/) chooses the primary key to make it easier for people to switch libraries. I believe they need _both_ [Key] and [Required], so it's not quite the same.

Is this preventing you from doing something? The best way to show me what you want would be to add it as a test in the ConsoleTests project or make a [Gist](https://gist.github.com/).

**Comment by ffhighwind** [comment]
This change will probably break people's code, but I see your point. I've added these to ConsoleTests as Dto14 and Dto15. Let me know if this fixes your problem. I will push it to the nuget package in a few days.

**Comment by ShadyHarbinger** [comment]
Hi ffhighwind. I intend to check the fix tomorrow. Till then here is an example model that could have been generated with Entity Framework and ExtraCRUD could not handle correctly:

`    public class ExampleModel
    {
        public int ID; // Primary autoincrement key
        [Required]
        public int RelationID; // Not primary key (ID of another record in another table)
        [Required]
        public int BookCount; // Cannot be null but it is not a key
    }
`

**Comment by ffhighwind** [comment]
The new update should support this, but only for properties. I dropped support for fields a while ago because Dapper only supports them if they're public. I could make protected/private fields work by always parsing an object into a DynamicParameters, but that would make me unhappy. If I added support for fields at this point then it will break some people's code, but I would consider doing it for public fields.
